subprojects/docs/src/docs/userguide/what_is_gradle.adoc

```diff
@@ -12,133 +12,114 @@
-
-Another important aspect of the configuration phase is that everything involved in it is evaluated _every time the build runs_. That is why it's best practice to <<authoring_maintainable_build_scripts#sec:minimize_logic_executed_configuration_phase,avoid expensive work during the configuration phase>>. https://scans.gradle.com/[Build scans] can help you identify such hotspots, among other things.
-
-=== 4. Gradle is extensible in more ways than one
-
-It would be great if you could build your project using only the build logic bundled with Gradle, but that's rarely possible. Most builds have some special requirements that mean you need to add custom build logic.
-
-Gradle provides several mechanisms that allow you to extend it, such as:
-
- * <<custom_tasks#custom_tasks,Custom task types>>.
-+
-When you want the build to do some work that an existing task can't do, you can simply write your own task type. It's typically best to put the source file for a custom task type in the <<organizing_gradle_projects#sec:build_sources,_buildSrc_>> directory or in a packaged plugin. Then you can use the custom task type just like any of the Gradle-provided ones.
-
- * Custom task actions.
-+
-You can attach custom build logic that executes before or after a task via the link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:doFirst(org.gradle.api.Action)[Task.doFirst()] and link:{groovyDslPath}/org.gradle.api.Task.html#org.gradle.api.Task:doLast(org.gradle.api.Action)[Task.doLast()] methods.
+=== Builds
 
- * <<writing_build_scripts#sec:extra_properties,Extra properties>> on projects and tasks.
-+
-These allows you to add your own properties to a project or task that you can then use from your own custom actions or any other build logic. Extra properties can even be applied to tasks that aren't explicitly created by you, such as those created by Gradle's core plugins.
+A build is an execution of a collection of tasks in a Gradle project. You instantiate a build with a Gradle command
+via the CLI or an IDE. In the command, you specify a task or tasks. Gradle determines the complete set of tasks to run based
+on inter-task dependencies, and runs the smallest set of tasks required to complete the specified task and its dependencies.
 
- * Custom conventions.
-+
-Conventions are a powerful way to simplify builds so that users can understand and use them more easily. This can be seen with builds that use standard project structures and naming conventions, such as <<building_java_projects#building_java_projects,Java builds>>. You can write your own plugins that provide conventions — they just need to configure default values for the relevant aspects of a build.
+== Optimizations
```

Comment:
Is this the right place for this? This chapter is "What is Gradle?".  Should this be in the performance chapter? Should this be under a "Next steps" or linked from Getting Started somehow?

Reply:
And should these sections be linking out to longer form chapters as well?

Reply:
I mostly meant to bring up important concepts that users will inevitably encounter even in their early days of Gradle usage, so they at least have some basic understanding of ideas like the configuration cache and the build cache. Do you feel like that's a good idea?

Reply:
The Optimizations section just seems out of place for a "What is Gradle?" chapter. 

The way I'm thinking about this is that we have people entering this page from the Getting Started or from The Google.  I think we want this page to just describe what Gradle is (short description, some terminology, some aspirations/vision) and nothing else, except some breadcrumbs to get to something else. 

I think some of the technical vision still belongs here somehow: https://github.com/gradle/gradle/pull/22277#pullrequestreview-1141502330 

I think as a stretch, I can see listing optimizations as part of the reason that Gradle's fast, but I don't think that's as important to mention here. There are lots of other things that are "optimizations" that could be covered here.

I think it could make sense here to call out things that someone might expect from a modern build system:
- Parallel execution
- Incremental execution
- JVM ABI compilation avoidance
- Local and remote caching
- File system watching

This could also provide a natural place to link to Gradle Enterprise and how GE augments some of these things, but that could be added later.

Reply:
Removed the section -- I don't think it belongs on this page any more.